diff --git a/target/linux/ramips/image/rt305x.mk b/target/linux/ramips/image/rt305x.mk
index 815e03a8d4..404db5b60c 100644
--- a/target/linux/ramips/image/rt305x.mk
+++ b/target/linux/ramips/image/rt305x.mk
@@ -390,6 +390,7 @@ endef
 TARGET_DEVICES += dlink_dir-300-b7
 
 define Device/dlink_dir-320-b1
+  $(Device/uimage-lzma-loader)
   SOC := rt5350
   IMAGE_SIZE := 7872k
   DEVICE_VENDOR := D-Link
